tests/unit/commands/heading-test.js

```diff
@@ -0,0 +1,142 @@
+import { Editor } from 'content-kit-editor';
+import Helpers from '../../test-helpers';
+import { MOBILEDOC_VERSION } from 'content-kit-editor/renderers/mobiledoc';
+import HeadingCommand from 'content-kit-editor/commands/heading';
+import SubheadingCommand from 'content-kit-editor/commands/subheading';
+
+const { test, module } = Helpers;
+
+let fixture, editor, editorElement, selectedText, command;
+
+const mobiledoc = {
+  version: MOBILEDOC_VERSION,
+  sections: [
+    [],
+    [[
+      1, 'P', [[[], 0, 'THIS IS A TEST']],
+      1, 'P', [[[], 0, 'second section']]
+    ]]
+  ]
+};
+
+module('Unit: HeadingCommand', {
+  beforeEach() {
+    fixture = document.getElementById('qunit-fixture');
+    editorElement = document.createElement('div');
+    editorElement.setAttribute('id', 'editor');
+    fixture.appendChild(editorElement);
+    editor = new Editor({mobiledoc});
+    editor.render(editorElement);
+    command = new HeadingCommand(editor);
+
+    selectedText = 'IS A';
+    Helpers.dom.selectText(selectedText, editorElement);
+    Helpers.dom.triggerEvent(document, 'mouseup');
+  },
+
+  afterEach() {
+    if (editor) { editor.destroy(); }
+  }
+});
+
+test('highlight text, click "heading" button turns text into h2 header', (assert) => {
+  const done = assert.async();
+
+  setTimeout(() => {
+    command.exec();
+    assert.hasElement('#editor h2:contains(THIS IS A TEST)');
+    assert.selectedText('THIS IS A TEST', 'expands selection to entire section');
+
+    done();
+  });
+});
+
+test('click heading button triggers update', (assert) => {
+  const done = assert.async();
+  const triggered = [];
+  const triggerFn = editor.trigger;
+  editor.trigger = (name, ...args) => {
+    triggered.push(name);
+    triggerFn.call(editor, name, ...args);
+  };
+
+  setTimeout(() => {
```

Comment:
same as above, I do not think that there is async involved in this code path any longer